src/blocklypropclient.js

```diff
@@ -335,33 +296,31 @@ var configure_client = function () {
 function establish_socket() {
 
     // TODO: set/clear and load buttons based on status
-    if (!client_available) {
+    if (!clientService.available) {
 
         // Clear the port list
         set_port_list();
 
-        var url = "ws://" + client_domain_name + ":" + client_domain_port + "/";
-        var connection = new WebSocket(url);
+        var connection = new WebSocket(clientService.url('ws'));
 
         connection.onopen = function () {
 
-            if (client_ws_connection !== null) {
-                client_ws_connection.close();
+            if (clientService.activeConnection !== null) {
+                clientService.activeConnection.close();
             }
 
             var ws_msg = {type: 'hello-browser', baud: baudrate};
-            client_ws_connection = connection;
+            clientService.activeConnection = connection;
             connection.send(JSON.stringify(ws_msg));
         };
 
         // Log errors
         connection.onerror = function (error) {
             // Only display a message on the first attempt
-            if (client_use_type !== 'ws' && !check_ws_socket_timeout) {
+            if (!clientService.type && !check_ws_socket_timeout) {
                 console.log('Unable to find websocket client');
             } else {
                 console.log('Websocket Communication Error');
```

Comment:
After a bunch of research, there is no way (not even `try {} catch () {}`) to eliminate the red error in the console when a websocket does not connect.  Guess we will just have to live with it :(